### ISSUE TITLE
Sign in – Able to dismiss Onboarding modal when click on browser back button

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#47246]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.19-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**Email or phone of affected tester (no customers):** ponikarchuks+1812824@gmail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Go to https://staging.new.expensify.com/
2. In Sign in page enter new Gmail account
3. When Onboarding modal appears click on browser back button

## Expected Result:
Onboarding modal is present

## Actual Result:
Onboarding modal disappears

## Workaround:
Unknown

## Platforms:

- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/a463e86e-280b-4fab-a537-9524d81653b8

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by IuliiaHerets** [comment]
We think that this bug might be related to #wave-collect

**Comment by mountiny** [comment]
@filip-solecki Could you have a look into this in the morning please?

**Comment by Beamanator** [comment]
Should we call this NAB if we're ok waiting till morning for a resolution? 🤔

**Comment by Beamanator** [comment]
Agreed! Marking NAB!±

**Comment by filip-solecki** [comment]
Me and @adamgrzybowski  are not able to reproduce it, did you try to clear Onyx data?

**Comment by mountiny** [comment]
Asked in QA if they can repro this on mac too https://expensify.slack.com/archives/C9YU7BX5M/p1723547272644439 the video is from windows

**Comment by adamgrzybowski** [comment]
https://github.com/user-attachments/assets/d2d566ba-28ce-4e46-8a16-59160f8a89e1

As you can see I can't reproduce it on my end. The difference I noticed in my and your case is that the URL with the report id shows briefly in your. 

Not sure why this happens but I won't be able to debug this issue without reliable reproduction steps. 

I can point you to this function https://github.com/Expensify/App/blob/c5feb89f9f02cfb303e86246408f40c4f5a23119/src/libs/Navigation/AppNavigator/createCustomStackNavigator/CustomRouter.ts#L103-L118

I would guess this value `const targetFocusedRoute = findFocusedRoute(action?.payload); ` is wrong. Could you please check it with the debugger? Maybe I will see what is wrong

**Comment by deetergp** [comment]
@kavimuru Were your reproduction steps on Mac any different than what is listed in the GH description?

**Comment by adamgrzybowski** [comment]
This may be a race condition. It looks like the initial chat doesn't have time to load and break the behavior on my end.

**Comment by deetergp** [comment]
I reached out to @kavimuru in Slack.

**Comment by deetergp** [comment]
I asked for a retest in Slack but have not yet heard back.

**Comment by trjExpensify** [comment]
> I asked for a retest in Slack but have not yet heard back.

@deetergp can you link the thread when making these kind of references, just helps follow the breadcrumbs. Where are we at with this re-test?

**Comment by deetergp** [comment]
Of course! This is where I asked previously https://expensify.slack.com/archives/C9YU7BX5M/p1724783266169849, but I think it got lost in the shuffle. Asked again here https://expensify.slack.com/archives/C9YU7BX5M/p1725431001409449

**Comment by m-natarajan** [comment]
Not reproducible

https://github.com/user-attachments/assets/55d61b2d-79c1-44aa-bedd-3f411f4983c2

**Comment by trjExpensify** [comment]
Cool, so then let's close this out @deetergp?
